### PR TITLE
Fix `tf.keras.initializers.get`: convert provided type object to object

### LINKED ISSUE #47054: tf.keras.initializers.zeros causes model.save to fail, while tf.keras.initializers.Zeros() works great
[Here](https://www.tensorflow.org/api_docs/python/tf/keras/initializers/Zeros) it's written that `tf.keras.initializers.zeros` is a shortcut for `tf.keras.initializers.Zeros()`.
If it is a shortcut, then both should work same

While saving the model, if I use `tf.keras.initializers.zeros` , model save failes, but using `tf.keras.initializers.Zeros()` works great.

Same issue raised by someone at [Stackoverflow](https://stackoverflow.com/questions/57154799/keras-model-saving-erroring-typeerror-get-config-missing-1-required-position) 

[Colab](https://colab.research.google.com/drive/1E0P-aBU9B7RO_QUtDrlRDPcOWqd6UfkD?usp=sharing#scrollTo=weBjeZAFJOP4)

### PR DESCRIPTION
**Pull request by mishc9**
Closes #47054

Summary: 

Models with layers instantiated with initializers provided as type objects couldn't be saved after training, despite models were initialized and trained fine ([colab](https://colab.research.google.com/drive/1ehWxx6PaDKP6nFMaI2Nf84O3-Xyv3-Sm?usp=sharing) with example).  This problem have been resolved the same way as in the `add_weight` method: now type objects are converting to objects in the function `initializers.get`. A unit test covering this case was implemented.

Possible alternative fixes:
* Remove support for type object initializers from `add_weight`. Drawback: would break backward compatibility
* Make any initializer class (not just object) serializable. Could work, but hard to maintain and confusing.
* Add instantiation of type objects of initializer type to the serialization function. Drawbacks: this function affects many other things and we should not overcomplicate it. Also it could lead to bugs in the parts of the code base non-related to initializets.

### CODE REVIEW

**Review by mattdangerw — APPROVED** [review]
Thanks for the fix!

**Comment by mishc9** [comment]
CI failed due to my mistype, unfortunately. I am new to the Tensorflow CI, and still couldn't manage to run TF unit tests locally. So, I've fixed mistype which causes an error, but not sure if I should run tests locally before you start them in the Google CI.

**Review by mattdangerw — APPROVED** [review]
Typo fix looks good.

**Comment by rthadur** [comment]
@mishc9 can you please check sanity build failures ?

**Comment by mishc9** [comment]
@rthadur I've checked the failures and fixed bad indentation in my PR. Thank you for pointing that.

**Review by rthadur — APPROVED** [review]

File: tensorflow/python/keras/utils/generic_utils_test.py [hunk 1]
```diff
@@ -310,6 +310,21 @@ def from_config(cls, config):
...
     self.assertIs(new_layer.units.fn, serializable_fn)
 
+  def test_serialize_type_object_initializer(self):
+    layer = keras.layers.Dense(
+        1,
+        kernel_initializer=keras.initializers.ones,
+        bias_initializer=keras.initializers.zeros)
+    config = keras.layers.serialize(layer)
+    self.assertEqual(
+        config['config']['bias_initializer'],
```
Comment (mattdangerw) [inline tensorflow/python/keras/utils/generic_utils_test.py:320]: Actually, in certain conditions we will stick a dtype into the config, causing this to fail. Could you update this to 

config['config']['bias_initializer']['class_name'], 'Zeros'

That should be more robust and allow us to merge this.

File: tensorflow/python/keras/utils/generic_utils_test.py [hunk 2]
```diff
@@ -310,6 +310,21 @@ def from_config(cls, config):
...
+        {'class_name': 'Zeros', 'config': {}}
+    )
+    self.assertEqual(
+        config['config']['kernel_initializer'],
```
Comment (mattdangerw) [inline tensorflow/python/keras/utils/generic_utils_test.py:324]: Here too

**Review by mattdangerw — COMMENTED** [review]
Hit a block to getting this merge, but should be a simple fix. Thanks!

File: tensorflow/python/keras/utils/generic_utils_test.py (hunk 2 above)
Comment (mishc9) [reply tensorflow/python/keras/utils/generic_utils_test.py]: Done

**Review by rthadur — APPROVED** [review]

**Pull request merged**